src/Utilities/UnitTests/CommandLineBuilder_Tests.cs

```diff
@@ -405,19 +405,19 @@ public void AppendSwitchUnQuotedTaskItem()
         }
 
         /// <summary>
-        /// Odd number of literal quotes. This should trigger an exception, because command line parsers
-        /// generally can't handle this case.
+        /// Ensure it's not an error to have an odd number of literal quotes. Sometimes
+        /// it's a mistake on the programmer's side, but we cannot reject odd numbers of
+        /// quotes in the general case because sometimes that's exactly what's needed (e.g.
+        /// passing a string with a single embedded double-quote to a compiler).
         /// </summary>
         [Fact]
         public void AppendSwitchWithOddNumberOfLiteralQuotesInParameter()
         {
-            Assert.Throws<ArgumentException>(() =>
-            {
-                CommandLineBuilder c = new CommandLineBuilder();
-                c.AppendSwitchIfNotNull("/D", @"ASSEMBLY_KEY_FILE=""c:\\foo\\FinalKeyFile.snk");
-            }
-           );
+            CommandLineBuilder c = new CommandLineBuilder();
+            c.AppendSwitchIfNotNull("/D", @"A='""'");
+            Assert.Equal(@"/D""A='\""'""", c.ToString());
```

Comment:
Quote escaping rules here are hurting my brain! For my own edification:

`@"A='""'" == "A='\"'"` is `A='"'`

`@"/D""A='\""'""" == "/D\"A='\\\"'\""` is `/D"A='\"'"`

Which makes sense, since `"` is in the [`definitelyNeedQuotesRegexNoHyphen`](https://github.com/Microsoft/msbuild/blob/70db3602ec75e0e4c50b613eb6588e158d5dfd9f/src/Utilities/CommandLineBuilder.cs#L135) list.